### ISSUE TITLE
Implement Weight Decay Functions

### ISSUE THREAD

**Issue by bstadt** [bstadt/neural_net_from_scratch#3]

**Comment by bstadt** [comment]
Current workaround is exposing learning rate as a public variable in graph.
